### DISCUSSION TITLE
Does NATS support JWT-SVID authentication?

### DISCUSSION THREAD

**Discussion by AlexHu0208** [nats-io/nats-server#5495] [Q&A] [answered]
I am wondering if it is possible to use SPIFFE [JWT-SVIDs](https://github.com/spiffe/spiffe/blob/main/standards/JWT-SVID.md) for NATS authentication.  Can we use the SPIFFE/SPIRE generated identities as the NATS identities?

Highly appreciate your help.

**Comment by wallyqs** [comment]
yes, there is basic support for this in the server: https://github.com/nats-io/nats-server/blob/f90ef7be44aa3c97a20e34bb1ad01f0940dfbb0f/test/tls_test.go#L1704-L1742

**Reply by AlexHu0208** [reply]
Thanks for the quick reply. I have 2 follow-up questions
1. I can still use the same private key as what I did for the non-SPIFFE/SPIRE JWT to connect to the NATS server, correct?
```
nc, err := nats.Connect(url, nats.UserJWT(jwtCB, sigCB))
```
2. For the authorization, do I need to inject the pub/sub claims into the SPIRE?

I am still learning the SPIFFE/SPIRE and JWT-SVID, hope these questions make sense.

**Reply by wallyqs** [reply, marked as answer]
ah sorry my mistake, no there is no support for JWT-SVIDs at the moment, only basic support for X509-SVID:
https://github.com/spiffe/spiffe/blob/main/standards/X509-SVID.md

**Reply by AlexHu0208** [reply]
No worries! Thanks for the clarification! Is there any plan to support JWT-SVID in the near future?
